packages/core/contracts/financial-templates/common/financial-product-libraries/long-short-pair-libraries/RangeBondLongShortPairFinancialProductLibrary.sol

```diff
@@ -51,59 +51,56 @@ contract RangeBondLongShortPairFinancialProductLibrary is LongShortPairFinancial
      * c) highPriceRange > lowPriceRange.
      * d) parameters price can only be set once to prevent the deployer from changing the parameters after the fact.
      * e) For safety, a parameters should be set before depositing any synthetic tokens in a liquidity pool.
-     * f) financialProduct must expose an expirationTimestamp method to validate it is correctly deployed.
+     * f) longShortPair must expose an expirationTimestamp method to validate it is correctly deployed.
      */
     function setLongShortPairParameters(
-        address LongShortPair,
+        address longShortPair,
         uint256 highPriceRange,
         uint256 lowPriceRange
     ) public nonReentrant() {
-        require(ExpiringContractInterface(LongShortPair).expirationTimestamp() != 0, "Invalid LSP address");
+        require(ExpiringContractInterface(longShortPair).expirationTimestamp() != 0, "Invalid LSP address");
 
         require(highPriceRange > lowPriceRange, "Invalid bounds");
 
-        RangeBondLongShortPairParameters memory params = longShortPairParameters[LongShortPair];
+        RangeBondLongShortPairParameters memory params = longShortPairParameters[longShortPair];
         require(params.highPriceRange == 0 && params.lowPriceRange == 0, "Parameters already set");
 
-        longShortPairParameters[LongShortPair] = RangeBondLongShortPairParameters({
+        longShortPairParameters[longShortPair] = RangeBondLongShortPairParameters({
             highPriceRange: highPriceRange,
             lowPriceRange: lowPriceRange
         });
     }
 
     /**
-     * @notice Returns a number between 0 and 1 to indicate how much collateral each long and short token are entitled
+     * @notice Returns a number between 0 and 1e18 to indicate how much collateral each long and short token are entitled
      * to per collateralPerPair.
      * @param expiryPrice price from the optimistic oracle for the LSP price identifier.
      * @return expiryPercentLong to indicate how much collateral should be sent between long and short tokens.
      */
-    function computeExpiryTokensForCollateral(int256 expiryPrice)
+    function percentageLongCollateralAtExpiry(int256 expiryPrice)
         public
         view
         override
         nonReentrantView()
         returns (uint256)
     {
         RangeBondLongShortPairParameters memory params = longShortPairParameters[msg.sender];
+        require(params.highPriceRange != 0 && params.lowPriceRange != 0, "Params not set for calling LSP");
 
-        // expiryPercentLong=[min(max(1/R2,1/P),1/R1)]/(1/R1)
```

Comment:
OZ comment on simplifying some logic in this computation:
The `computeExpiryTokensForCollateral` function of the `RangeBondLongShortPairFinancialProductLibrary` uses [a complicated closed-form expression](https://github.com/UMAprotocol/protocol/blob/b508f536ddfd94a79f93f633142bdc868b73461c/packages/core/contracts/financial-templates/common/financial-product-libraries/long-short-pair-libraries/RangeBondLongShortPairFinancialProductLibrary.sol#L82) to determine how much collateral to assign to the long position. Instead, consider enumerating the cases explicitly so it is easier to reason about. This would reduce to something like:
```solidity
// (collateral tokens * lowPriceRange) is the notional value of the bond
// below the range, the collateral is worth less than this amount
// the long position is entitled to whatever is left
if(expiryPrice <= lowPriceRange) return 1;
// within the range, the long position is entitled to the notional value, 
// which is equal to the following fraction of collateral
if(expiryPrice <= highPriceRange) return lowPriceRange / expiryPrice;
// above the range, the long position is entitled to a fixed number of tokens
return lowPriceRange / highPriceRange;
```

Reply:
+1 for including this comment. Wasn't entirely sure at first what was going on but i figured you just refactored for readability

Reply:
100%. This form is the same that I showed at the end of the LSP presentation a few weeks ago where you transform the equation into a piecewise function that looks something like this:
![image](https://user-images.githubusercontent.com/12886084/123218928-9a2d2980-d4cc-11eb-9e9b-397c64c69725.png)